cub/test/catch2_test_device_reduce_large_offsets.cu

```diff
@@ -0,0 +1,247 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "insert_nested_NVTX_range_guard.h"
+// above header needs to be included first
+
+#include <cub/device/device_reduce.cuh>
+#include <cub/thread/thread_operators.cuh>
+
+#include <thrust/iterator/counting_iterator.h>
+
+#include <cuda/std/__algorithm_>
+
+#include <cstdint>
+
+#include "catch2_test_device_reduce.cuh"
+#include "catch2_test_helper.h"
+#include "catch2_test_launch_helper.h"
+
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::Reduce, device_reduce);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::Sum, device_sum);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::Min, device_min);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::ArgMin, device_arg_min);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::Max, device_max);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::ArgMax, device_arg_max);
+
+// %PARAM% TEST_LAUNCH lid 0:1:2
+
+// List of offset types to test
+using offset_types = c2h::type_list<std::int32_t, std::uint32_t, std::uint64_t>;
```

Comment:
Question: Why is `int64_t` not included?

Reply:
It's just a trade-off between test coverage and increased compilation and test time.

Reply:
Hmm, why do test 32-bit offset types at all? We already have tests for `int32` in `catch2_test_device_reduce.cu`.

Reply:
I like to think of these tests as "testing large problem sizes". The verification of these tests should be quite efficient and add only little time to our total testing time, while giving us the benefit of testing for integer overflow when running close to `INT_MAX` number of items.